### PR TITLE
[cmake] use treedata instead of add_directory for omxplayer and rpi windowing

### PR DESCRIPTION
**Pull request by Rechi**
## Motivation and Context
unify adding directories for building to the build system

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by lrusak — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -296,6 +296,7 @@ set(OTHER_FILES cmake/README.md)
 
 # Subdirs
 core_add_subdirs_from_filelist(${CMAKE_SOURCE_DIR}/cmake/treedata/common/*.txt
+                               ${CMAKE_SOURCE_DIR}/cmake/treedata/common/${CORE_PLATFORM_NAME_LC}/*.txt
```
Comment (fuzzard) [inline CMakeLists.txt]: Just a query. Do they really belong in treedata/common/${core_platform_name_lc} or should they be put into treedata/${core_system_name}/${core_platform_name_LC} ?

Comment (wsnipex) [reply CMakeLists.txt]: I was thinking the same. Why in common?

Comment (Rechi) [reply CMakeLists.txt]: aml, gbm, rbpi, wayland and x was already handled in common before. So nothing has changed.
The reason is, they are needed for linux and freebsd.

**Comment by FernetMenta** [comment]
omxplayer was supposed to be dropped after Leia: https://github.com/xbmc/xbmc/pull/13592#issuecomment-368915709

might have changed though

**Pull request merged**